Framework/API/inc/MantidAPI/IndexProperty.h

```diff
@@ -36,7 +36,7 @@ class MANTID_API_DLL IndexProperty : public Kernel::ArrayProperty<int64_t> {
                     Kernel::IValidator_sptr(new Kernel::NullValidator));
 
   IndexProperty(const IndexProperty &) = default;
-  IndexProperty &operator=(const IndexProperty &) = default;
+  IndexProperty &operator=(const IndexProperty &) = delete;
```

Comment:
Do we understand what causes the copy-assignment to be deleted but not the copy constructor?

Reply:
This is the original warning, not sure however why copy constructor is different
```
IndexProperty.h:39:18: warning: explicitly defaulted copy assignment operator is implicitly deleted
IndexProperty.h:58:29: note: copy assignment operator of 'IndexProperty' is implicitly deleted because field 'm_workspaceProp' is of reference type 'const Mantid::API::IWorkspaceProperty &'
```

Reply:
Ah I see. As you can't rebind a reference then any class that contains a reference-type member cannot be assigned and so it is [deleted](https://en.cppreference.com/w/cpp/language/copy_assignment#Deleted_implicitly-declared_copy_assignment_operator). There is not the same restriction for copy constructors as I guess it's initialization of a new object from an existing reference.

Reply:
The question is if it is deleted implicitly, does it make sense to delete explicitly, or we can as well skip it?

Reply:
I guess this is something of a preference. It does feel like this isn't as obvious why one is deleted and not the other so perhaps explicit is better. Perhaps we should add a comment as it does violate the rule of 0/3/5 and is unexpected?